File: kiranaos-store-service/src/main/java/com/kiranaos/kiranaos_store_service/domain/Product.java
package com.kiranaos.kiranaos_store_service.domain;

import com.kiranaos.kiranaos_store_service.domain.enums.UnitType;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Id;
import jakarta.persistence.GenerationType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Column;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.EnumType;
import jakarta.persistence.FetchType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "products")
@Setter
@Getter
@NoArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", nullable = false)
    private Store store;

    @Column(name = "store_id", insertable = false, updatable = false)
    private UUID storeId;

    private String category;

    @Enumerated(EnumType.STRING)
    private UnitType unit;

    @Column(nullable = false)
    private BigDecimal price;

    private BigDecimal gstRate;

    @Column(nullable = false)
    private BigDecimal stockQuantity;

    private BigDecimal reorderThreshold;

    @Column(nullable = false)
    private Boolean isDeleted = false;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;
}

File: kiranaos-web/src/types/product.ts
export type UnitType = 'KG' | 'PIECE' | 'LITRE' | 'PACKET';

export interface CreateProductRequest {
  name: string;
  category: string;
  unit: UnitType;
  price: number;
  gstRate?: number;
  stockQuantity: number;
  reorderThreshold?: number;
}

export interface UpdateProductRequest {
  name?: string;
  category?: string;
  unit?: UnitType;
  price?: number;
  gstRate?: number;
  stockQuantity?: number;
  reorderThreshold?: number;
}

export interface ProductResponse {
  id: string;
  storeId: string;
  name: string;
  category: string;
  unit: UnitType;
  price: number;
  gstRate: number;
  stockQuantity: number;
  reorderThreshold: number;
}

export interface ProductImportResponse {
  successCount: number;
  failedCount: number;
  failedReasons: string[];
}
